src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -347,6 +349,55 @@ public void setInSelectCartesianProductThreshold(int warn, int fail)
                                   x -> config.in_select_cartesian_product_fail_threshold = x);
     }
 
+    public DataStorageSpec getCollectionSizeWarnThreshold()
+    {
+        return config.collection_size_warn_threshold;
+    }
+
+    @Override
+    public DataStorageSpec getCollectionSizeFailThreshold()
+    {
+        return config.collection_size_fail_threshold;
+    }
+
+    public void setCollectionSizeThreshold(DataStorageSpec warn, DataStorageSpec fail)
+    {
+        validateSizeThreshold(warn, fail, "collection_size");
```

Comment:
Do iiuc?:
- This won't check for negative sizes?
- The code in the method returns if _either_ of the 2 is disabled when we should check for positive sizes on _both_

Reply:
lgtm